packages/wonder-blocks-button/src/components/button.tsx

```diff
@@ -240,65 +60,43 @@ const Button: React.ForwardRefExoticComponent<
         inRouterContext,
     );
 
-    const renderProp = (
-        state: ClickableState,
-        restChildProps: ChildrenProps,
-    ) => {
-        return (
-            <ThemedButton>
-                <ButtonCore
-                    {...sharedButtonCoreProps}
-                    {...state}
-                    {...restChildProps}
-                    disabled={disabled}
-                    spinner={spinner || state.waiting}
-                    actionType={actionType}
-                    kind={kind}
-                    size={size}
-                    skipClientNav={skipClientNav}
-                    href={href}
-                    target={target}
-                    type={type}
-                    tabIndex={tabIndex}
-                    ref={ref}
-                >
-                    {children}
-                </ButtonCore>
-            </ThemedButton>
-        );
-    };
-
-    if (beforeNav) {
-        return (
-            <ClickableBehavior
-                disabled={spinner || disabled}
-                href={href}
-                role="button"
-                type={type}
-                onClick={onClick}
-                beforeNav={beforeNav}
-                safeWithNav={safeWithNav}
-                rel={rel}
-            >
-                {renderProp}
-            </ClickableBehavior>
-        );
-    } else {
-        return (
-            <ClickableBehavior
-                disabled={spinner || disabled}
-                href={href}
-                role="button"
-                type={type}
-                onClick={onClick}
-                safeWithNav={safeWithNav}
-                target={target}
-                rel={rel}
-            >
-                {renderProp}
-            </ClickableBehavior>
-        );
-    }
+    const extraClickableProps = beforeNav ? {beforeNav} : {target};
+
+    return (
+        <ClickableBehavior
+            disabled={spinner || disabled}
+            href={href}
+            role="button"
```

Comment:
Do we want it to have `role="link"` if it is a link and has a `href`? (Previously, ClickableBehavior always had `role="button"` so this was pre-existing!)

Or do we need to set the role? We choose between a `<button>` or `<a>` in the `ButtonUnstyled` component! 

Here is the behaviour where an anchor tag is used with `role=button`:
![image](https://github.com/user-attachments/assets/f10f7219-3c5d-4300-8247-c3c7632159df)

Reply:
That makes total sense, although I'd prefer to hold on this change for a future PR. I'd like to avoid introducing any unintended behaviors with this change 😬. I'll create a ticket to track this.

Reply:
Added as a TODO comment: https://khanacademy.atlassian.net/browse/WB-1940